thetis/shallowwater_eq.py

```diff
@@ -341,13 +363,13 @@ def residual(self, uv, eta, uv_old, eta_old, fields, fields_old, bnd_conditions)
             for bnd_marker in self.boundary_markers:
                 funcs = bnd_conditions.get(bnd_marker)
                 ds_bnd = ds(int(bnd_marker), degree=self.quad_degree)
-                if funcs is not None:
+                if self.bnd_state_defined(funcs, bnd_marker):
                     eta_ext, uv_ext = self.get_bnd_functions(head, uv, bnd_marker, bnd_conditions)
                     # Compute linear riemann solution with eta, eta_ext, uv, uv_ext
                     un_jump = inner(uv - uv_ext, self.normal)
                     eta_rie = 0.5*(head + eta_ext) + sqrt(total_h/g_grav)*un_jump
                     f += g_grav*eta_rie*dot(self.u_test, self.normal)*ds_bnd
-                if funcs is None or 'symm' in funcs:
+                else:
```

Comment:
I take it we weren't actually using `symm`?

Reply:
yes, this feature was already dysfunctional as it would have raised an error in `get_bnd_functions`.